pkg/kubelet/kubelet.go

```diff
@@ -1751,13 +1751,41 @@ func (kl *Kubelet) getPullSecretsForPod(pod *api.Pod) ([]api.Secret, error) {
 	return pullSecrets, nil
 }
 
+// Return name of a volume. When the volume is a PersistentVolumeClaim,
+// it returns name of the real PersistentVolume bound to the claim.
+// It returns errror when the clam is not bound yet.
+func (kl *Kubelet) resolveVolumeName(pod *api.Pod, volume *api.Volume) (string, error) {
+	claimSource := volume.VolumeSource.PersistentVolumeClaim
+	if claimSource != nil {
+		// resolve real volume behind the claim
+		claim, err := kl.kubeClient.PersistentVolumeClaims(pod.Namespace).Get(claimSource.ClaimName)
```

Comment:
`cleanupOrphanedVolumes` was blocking even before, detaching a volume usually takes a second or two.

I added a comment about that and filled https://github.com/kubernetes/kubernetes/issues/19645